experiments/exp1403_rl_math.py

```diff
@@ -0,0 +1,287 @@
+"""
+RL training experiment following marin patterns.
+"""
+
+import json
+import logging
+import os
+from dataclasses import dataclass
+
+import ray
+
+from marin.execution.executor import ExecutorStep, executor_main, this_output_path
+from marin.resources import ResourceConfig, TpuPodConfig
+from marin.training.training import (
+    _add_default_env_variables,
+    _add_run_env_variables,
+    _check_for_wandb_key,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class RlTrainOnPodConfig:
+    """Configuration for RL training on a pod, following marin's TrainLmOnPodConfig pattern."""
+
+    resources: ResourceConfig
+
+    load_model: str
+    sharding: str = "1,4,1,-1"
+    num_train_steps: int = 2048
+    max_input_length: int = 256
+    max_output_length: int = 1025
+    train_bsize: int = 64
+    decode_bsize: int = 1024
+    prefill_bsize: int = 16
+    reference_logprobs_bsize: int = 256
+    n_prompts_per_step: int = 16
+    log_freq: int = 8
+    num_eval_examples: int = 1024
+    save_model_freq: int = 0
+    wandb_project: str = "math_rloo_math_test_experiments"
+
+    inference_param_dtype: str = "bf16"
+    inference_activation_dtype: str = "bf16"
+    training_param_dtype: str = "fp32"
+    training_activation_dtype: str = "bf16"
+
+    optim_config: str = (
```

Comment:
Would be nice if we didn't have to replicate all of this - this should be in a utility file for llama specifically